MDANSE_GUI/Src/MDANSE_GUI/Widgets/PlotSettingsDialog.py

```diff
@@ -0,0 +1,339 @@
+
+        self.setWindowTitle("MDANSE Plot Settings Editor")
+
+        layout = QVBoxLayout(self)
+
+        self.setLayout(layout)
+        self.viewer = QTreeView(self)
+        self.viewer.setAnimated(True)
+        self.filter_combo = QComboBox(self)
+        self.filter_box = QLineEdit(self)
+        layout.addWidget(self.filter_combo)
+        mid_hbox = QHBoxLayout(self)
+        mid_hbox.addWidget(QLabel("Filter entries by:"))
+        mid_hbox.addWidget(self.filter_box)
+        mid_hbox.addWidget(QLabel("Set filter to keyword:"))
+        mid_hbox.addWidget(self.filter_combo)
+        layout.addLayout(mid_hbox)
+        layout.addWidget(self.viewer)
+        self._changed_keys = {}
+        self.load_settings()
+        self.model = PlotSettingsModel()
+        self.filter_proxy_model = QSortFilterProxyModel()
+        self.filter_proxy_model.setSourceModel(self.model)
+        self.viewer.setModel(self.filter_proxy_model)
+        self.filter_combo.addItems(["all"] + SORT_MAJOR_KEYS)
+        self.filter_combo.setCurrentText("all")
+        self.filter_combo.currentTextChanged.connect(self.filter_box.setText)
+        self.filter_box.textChanged.connect(self.filter_entries)
+
+        self.writeout_button = QPushButton("Save settings", self)
+        self.reset_button = QPushButton("Reset values", self)
+        layout.addWidget(self.writeout_button)
+        layout.addWidget(self.reset_button)
+        self.writeout_button.clicked.connect(self.save_changes)
+        self.reset_button.clicked.connect(self.reset_values)
+        self.viewer.expanded.connect(self.expand_columns)
+        self.viewer.resizeColumnToContents(0)
+        self.settings = settings
+        self.model.itemChanged.connect(self.register_item_change)
+        self.model.plots_need_updating.connect(self.values_changed)
+
+    @Slot(str)
+    def filter_entries(self, search_string: str):
+        """Hide rcParams keys which do not contain the search string.
+
+        Parameters
+        ----------
+        search_string : str
+            String to be found in valid dictionary keys.
+        """
+        if search_string == "all":
+            self.filter_proxy_model.setFilterFixedString("")
+        else:
+            self.filter_proxy_model.setFilterFixedString(search_string)
+
+    @Slot()
+    def expand_columns(self):
+        """Resize columns to the current amount of text in items."""
+        for ncol in range(3):
+            self.viewer.resizeColumnToContents(ncol)
+
+    def load_settings(self):
+        """Load previously saved plot settings from an MDANSE config file."""
+        settings_file = PLATFORM.application_directory() / "matplotlib.txt"
+        if not settings_file.exists():
+            LOG.info(
+                "File %s does not exist. Using standard matplotlib settings.",
+                str(settings_file),
+            )
+            return
+        rc_file(settings_file)
+        with open(settings_file) as source:
```

Comment:
```suggestion
        with settings_file.open(encoding="utf-8") as source:
```
Is the [preferred open](https://docs.astral.sh/ruff/rules/builtin-open/) for guaranteed `Path`s.